src/guiguts/tools/jeebies.py

```diff
@@ -46,193 +64,580 @@ def check_for_jeebies_in_file(self) -> None:
 
-        return llim, rlim + 1, para[llim : rlim + 1]
+        return suspects_count
 
-    def find_in_dictionary(self, three_form: str) -> int:
+    def locate_file_line(
+        self,
+        rec_num: int,
+        hebe_start: int,
+        paragraph_start_line_numbers: list[int],
+        paragraph_line_boundaries: list[list[int]],
+    ) -> tuple[int, int]:
+        """Returns line number in file and position on that line of a hebe found in paragraph string."""
+        paragraph_start_line_number = paragraph_start_line_numbers[rec_num]
+        hebe_start_position_on_line = 0
+        line_boundaries = paragraph_line_boundaries[rec_num]
+        for indx, boundary in enumerate(line_boundaries):
+            if hebe_start < boundary:
+                # Located file line number
+                line_number = paragraph_start_line_number + indx
+                if indx == 0:
+                    # We are on first line of paragraph
+                    hebe_start_position_on_line = hebe_start
+                else:
+                    hebe_start_position_on_line = hebe_start - line_boundaries[indx - 1]
+                break
+        return line_number, hebe_start_position_on_line
+
+    def find_in_dictionary(self, phrase: str) -> int:
         """Returns frequency in hebe phrase corpus or zero."""
 
-        words = three_form.split()
-        key = "|".join(words)
+        freq = self.dictionary.get(phrase, 0)
+        if freq != 0:
+            return freq
+        # Try again with a prefixing " " character
+        key = " " + phrase
+        freq = self.dictionary.get(key, 0)
+        if freq != 0:
+            return freq
+        # Try finally with a trailing " " character
+        key = phrase + " "
         freq = self.dictionary.get(key, 0)
-
         return freq
 
     def add_to_dialog(
-        self, header: str, para_slice: str, checker_dialog: CheckerDialog
+        self,
+        info: str,  # pylint: disable=unused-argument
+        line: str,
+        line_number: int,
+        hebe_start: int,
+        checker_dialog: CheckerDialog,
     ) -> None:
         """Helper function that abstracts repeated code.
 
         Args:
-            header: A 3-form phrase with explanatory info.
-            para_slice: A slice of a paragraph centered around the 3-form phrase.
+            info: Ratio of 'he' or 'be freq / 'be' or 'he' freq expressed as a string.
+            line: Text of file line.
+            line_num: Its line number in the file.
+            hebe_start: The position on line of string 'he' or 'be'.
             checker_dialog: Where report text is written.
         """
-
-        record = "    " + para_slice
-        # Add header and record to the dialog.
-        checker_dialog.add_entry(header)
-        checker_dialog.add_entry(record)
-        checker_dialog.add_entry(" ")
+        # Get start/end of 'he' or 'be' string in file.
+        error_start = str(line_number) + "." + str(hebe_start)
```

Comment:
f-strings might make this and the next line a bit cleaner as you don't need to cast the variables to strings and append them to each other, both are functionally the same so it's just an observation:
```python
error_start = f"{line_number}.{hebe_start}"
```

Reply:
I agree that in most cases using f-strings does make things 'cleaner'. The section of code you cite above is an example. It is now rendered as
```
        # Get start/end of 'he' or 'be' string in file.
        error_start = f"{line_number}.{hebe_start}"
        error_end = f"{line_number}.{hebe_start + 2}"
```
I have worked through the program making similar stylistic changes.

However there are a couple of places where I have used string concatenation combined with an f-string in order to be as obvious as possible about what is intended:
```
                # Not a blank line and previous line was not blank so
                # continuation of a paragraph
                s = s + f"{line} "
```